src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriterSettings.cs

```diff
@@ -917,5 +699,12 @@ internal XmlWriterSettings(XmlQueryDataReader reader)
             // bool isReadOnly;
             ReadOnly = reader.ReadBoolean();
         }
+
+        [DoesNotReturn]
+        [StackTraceHidden]
+        private static void ThrowArgumentOutOfRangeException(string paramName)
```

Comment:
why this wrapper? What's the gain of having this vs explicit throw?

Reply:
@krwq best codegen 
https://sharplab.io/#gist:2b8639b5e1e3c8dfd071d44b40f08693

Reply:
fair enough, I usually prefer explicit throws because they feel simpler - perhaps might be worth to ask someone in Roslyn to generate and call wrappers on compile for smaller code. @eerhardt do you think that could make any diff in the overall dll size?

Reply:
I can't imagine it making any significant size differences.

Although, similarly, I can't imagine this change making any measurable runtime perf differences as well. Since it is in property setters on XmlWriterSettings, which don't need to run in a tight loop in a real world app.